go/api/pubsub_message_handler.go

```diff
@@ -0,0 +1,174 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+// #include "../lib.h"
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"sync"
+)
+
+var (
+	ErrPubSubPushInvalid       = errors.New("received invalid push: empty or in incorrect format")
+	ErrPubSubPushMissingKind   = errors.New("received invalid push: missing kind field")
+	ErrPubSubPushMissingValues = errors.New("received invalid push: missing values field")
+)
+
+type MessageCallbackError struct {
+	cause error
+}
+
+func (e *MessageCallbackError) Error() string {
+	return fmt.Sprintf("error in message callback: %v", e.cause)
+}
+
+func (e *MessageCallbackError) Cause() error {
+	return e.cause
+}
+
+// *** Message Handler ***
+
+type MessageHandler struct {
+	callback MessageCallback
+	context  any
+	queue    *PubSubMessageQueue
+}
+
+func NewMessageHandler(callback MessageCallback, context any) *MessageHandler {
+	return &MessageHandler{
+		callback: callback,
+		context:  context,
+		queue:    NewPubSubMessageQueue(),
+	}
+}
+
+func (handler *MessageHandler) handleMessage(message *PubSubMessage) error {
+	if handler.callback != nil {
+		defer func() {
+			if r := recover(); r != nil {
+				err, ok := r.(error)
+				if !ok {
+					err = fmt.Errorf("%v", r)
+				}
+				log.Println("panic in message callback", err.Error())
+			}
+		}()
+
+		handler.callback(message, handler.context)
+		return nil
+	}
+
+	handler.queue.Push(message)
```

Comment:
This should be in `else` block

Reply:
It's not really required. It will optimized out by the compiler regardless. I've made the change regardless.